packages/core/src/tree-view/tree.element.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  AriaBooleanAttributeValues,
+  arrayHead,
+  arrayTail,
+  baseStyles,
+  internalProperty,
+  isVisible,
+  nextInArray,
+  onAnyKey,
+  onKey,
+  previousInArray,
+  property,
+  querySlot,
+  querySlotAll,
+  syncProps,
+} from '@cds/core/internal';
+import { html, LitElement, PropertyValues } from 'lit-element';
+import { CdsTreeItem } from './tree-item.element.js';
+import { styles } from './tree.element.css.js';
+
+/**
+ * Tree view is a hierarchical component that gives users access to a hierarchical set of objects displayed in a the parent-child relationship.
+ *
+ * ```typescript
+ * import '@cds/core/tree-view/register.js';
+ * ```
+ *
+ * ```html
+ *  <cds-tree>
+ *    <cds-tree-item>1</cds-tree-item>
+ *    <cds-tree-item>2</cds-tree-item>
+ *    <cds-tree-item>3</cds-tree-item>
+ *  </cds-tree>
+ * ```
+ *
+ * @beta
+ * @element cds-tree
+ * @slot - Content slot for inside the tree
+ */
+export class CdsTree extends LitElement {
+  @property({ type: String, reflect: true })
+  role = 'tree';
+
+  @property({ type: Boolean, attribute: 'multi-select' })
+  multiSelect = false;
+
+  @property({ type: String, reflect: true, attribute: 'aria-activedescendant' })
+  ariaActiveDescendant: string;
```

Comment:
I have a bunch of [specs](https://github.com/vmware/clarity/blob/3aee6317fc65901d1de464c7e59f994a3dfe4cd0/packages/core/src/tree-view/tree.element.spec.ts) that ascertain that the ariaActiveDescandant is updated to the right value.

I am kind of torn because using `internalProperty` or `state` decorator forces the field to be `protected`, which makes me lose the ability to access them in my tests. I know that making a field public for tests' sake isn't always the best but I feel that this is a more robust test than just asserting that the update function was called.

What if I just annotated this field as `/** @private */` so it doesn't show up in docs? Or if you have a different suggestion I'm pretty open to it!

Reply:
You should be able to use the `@state` decorator as a public class property. It uses the regular property decorator under the hood with some extras. We use a public property with a `@state` decorator [here](https://github.com/vmware/clarity/blob/next/packages/core/src/alert/alert.element.ts#L150)

The analyzer treats any property marked with `@internalProperty`, `@state` or `/** @private */` as internal APIs regardless if it's a public TS property and won't make them available in the editor autocomplete or API docs. If the build/lint is preventing it let me know there might be something I misconfigured with the lit 2 update.

Reply:
Ah, I haven't rebased with lit 2 update yet since it's mid-review. I'll verify if it'll let me do `@state` + public after the update.

Reply:
Addressed with the rebase 👍